.githooks/pre-commit

```diff
@@ -1,4 +1,50 @@
 #!/usr/bin/env bash
 
-cargo make check-rs-format && \
-cargo make check-ts-format
+debug=0                         # Set to 1 to enable debugging, 0 to disable.
+do_check_rs=0
+do_check_ts=0
+
+check_rs_cmd="cargo make check-rs-format"
+check_ts_cmd="cargo make check-ts-format"
+if (( debug == 1 )); then
+    check_rs_cmd="echo check_rs ran"
+    check_ts_cmd="echo check_ts ran"
+fi
+
+git diff --name-only | grep '.rs$' > /dev/null 2>&1
+if [ $? -eq 0 ]; then
+    do_check_rs=1
+fi
+
+git diff --name-only --cached | grep '.rs$' > /dev/null 2>&1
```

Comment:
:question: This may be obvious, but bear with me :stuck_out_tongue: Why are these `git diff`s done twice, but the second time with the `--cached` flag? Isn't this redundant?

Reply:
`man git` :)

Reply:
`--cached` acts _only_ on staged changes.  Without it acts on unstaged changes (i.e. files that are in the index but not staged).